src/providers/backblaze/provider.ts

```diff
@@ -0,0 +1,175 @@
+import { ReadStream, createReadStream } from 'node:fs';
+import { Readable } from 'node:stream';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { env } from 'node:process';
+import { fetch as uFetch } from 'undici';
+import chalk from 'chalk';
+import cuid2 from '@paralleldrive/cuid2';
+import { S3Client } from '@aws-sdk/client-s3';
+
+import { updateAsset, Asset } from '../../assets.js';
+import { findBucket, createBucket, putBucketCors, putObject } from '../../utils/s3.js';
+import log from '../../utils/logger.js';
+
+export type BackblazeSpecifics = {
+  bucket?: string;
+  endpoint?: string;
+}
+
+// Why 11?
+//  - Reasonable id length visually in the src URL
+//  - Familiarity with the length of YouTube IDs
+//  - It would take more than 300 million buckets to have a 50% chance of a collision.
+//  - "These go to eleven" https://www.youtube.com/watch?v=F7IZZXQ89Oc
+const createId = cuid2.init({ length: 11 });
+
+let s3: S3Client;
+let bucketName: string;
+let endpoint: string;
+
+async function initS3() {
+  bucketName = env.BACKBLAZE_BUCKET ?? '';
+  endpoint = env.BACKBLAZE_ENDPOINT ?? '';
```

Comment:
these might merit options in the next.config.js withNextVideo() instead? or in addition...

Reply:
the bucket is optional btw. if no name is provided one is generated with the 11 char long cuid